.github/workflows/wheels.yml

```diff
@@ -0,0 +1,97 @@
+# Workflow to build wheels for upload to PyPI.
+# While this workflow is not active right now,
+# it will be used for Cython 3.1 after Python 2.7
+# support is dropped.
+name: Wheel Builder
+on:
+  schedule:
```

Comment:
Do we really want to build wheels every push and PR to master?

Reply:
Currently, wheel building is a release-only workflow. I think we can keep it that way.

Reply:
I would advocate for a weekly build in addition to working on release. From my work on the wheels for numpy/pandas, these things tend to break from time to time. 

In numpy in particular, we also can trigger builds using a label or a certain commit message, to run the wheel builder CI when a PR changes the build config. If Cython wants this, I can also port this over too.

Reply:
Sounds great. Let's have weekly builds. We can even archive the wheels, as long as the release upload keeps being limited to releases.

Also +1 for supporting at least a label as trigger. Being able to add it to build related PRs would be really handy.

Reply:
We've occasionally had hard to diagnose problems with the shared module and changes to `CyFunction` between releases. We might see that more with regular pre-release builds, especially if they're used by a big project like Numpy.

It's probably solvable though (we could add the hashes of the utility code files to the shared API name).

Reply:
> Sounds great. Let's have weekly builds. We can even archive the wheels, as long as the release upload keeps being limited to releases.

The scientific python packages have been uploading their nightlies to https://anaconda.org/scipy-wheels-nightly/repo. We can probably upload our wheels there too, if you want. I think @mattip has the API token for this.

> 
> Also +1 for supporting at least a label as trigger. Being able to add it to build related PRs would be really handy.

I've currently set the label to "Build System". LMK if you want to use another label.